features/step_definitions/bookmark_steps.rb

```diff
@@ -46,3 +52,31 @@
     fill_in("collection_names", :with => "#{collection_name}")
     click_button("Add")
 end
+
+Then /^the bookmark on "([^\"]*)" should have tag "([^\"]*)"$$/ do |title, tag|
+  work = Work.find_by_title(title)
+  bookmark = work.bookmarks.first
+  bookmark.reload
+  bookmark.tags.collect(&:name).include?(tag)
+end
+
+Then /^the cache of the bookmark on "([^\"]*)" should expire after I edit the bookmark tags$/ do |title|
+  work = Work.find_by_title(title)
+  bookmark = work.bookmarks.first
+  orig_cache_key = bookmark.cache_key
+  visit edit_bookmark_path(bookmark)
+  fill_in("bookmark_tag_string", with: "New Tag")
+  click_button("Update")
+  bookmark.reload
+  assert orig_cache_key != bookmark.cache_key, "Cache key #{orig_cache_key} matches #{bookmark.cache_key}."
+end
+
+Then /^the cache of the bookmark on "([^\"]*)" should not expire if I have not edited the bookmark$/ do |title|
+  work = Work.find_by_title(title)
+  bookmark = work.bookmarks.first
+  orig_cache_key = bookmark.cache_key
+  visit edit_bookmark_path(bookmark)
+  visit bookmark_path(bookmark)
+  bookmark.reload
+  assert orig_cache_key == bookmark.cache_key, "Cache key #{orig_cache_key} does not match #{bookmark.cache_key}."
+end
```

Comment:
It's one of those historical best practice things but since Github cares enough to flag it too, I suppose there's no harm in making sure there are indeed newlines at the end of files: https://robots.thoughtbot.com/no-newline-at-end-of-file